core/include/traccc/sanity/contiguous_on.hpp

```diff
@@ -8,59 +8,52 @@
 
 #pragma once
 
-// VecMem include(s).
-#include <memory>
-#include <vecmem/containers/data/vector_view.hpp>
-#include <vecmem/containers/device_vector.hpp>
-#include <vecmem/memory/memory_resource.hpp>
-#include <vecmem/utils/copy.hpp>
-
 // System include
 #include <concepts>
 #include <memory>
 #include <unordered_set>
 
 namespace traccc::host {
+
 /**
- * @brief Sanity check that a given vector is contiguous on a given projection.
+ * @brief Sanity check that a given container is contiguous on a given
+ *        projection.
  *
- * For a vector $v$ to be contiguous on a projection $\pi$, it must be the case
- * that for all indices $i$ and $j$, if $v_i = v_j$, then all indices $k$
+ * For a container $c$ to be contiguous on a projection $\pi$, it must be the
+ * case that for all indices $i$ and $j$, if $v_i = v_j$, then all indices $k$
  * between $i$ and $j$, $v_i = v_j = v_k$.
  *
  * @note This function runs in O(n^2) time.
  *
  * @tparam P The type of projection $\pi$, a callable which returns some
  * comparable type.
- * @tparam T The type of the vector.
+ * @tparam CONTAINER The (SoA) container type
  * @param projection A projection object of type `P`.
  * @param mr A memory resource used for allocating intermediate memory.
  * @param vector The vector which to check for contiguity.
  * @return true If the vector is contiguous on `P`.
  * @return false Otherwise.
  */
-template <std::semiregular P, std::equality_comparable T>
-requires std::regular_invocable<P, T> bool is_contiguous_on(
-    P&& projection, vecmem::data::vector_view<T> vector) {
-    // Grab the number of elements in our vector.
-    uint32_t n = vector.size();
+template <std::semiregular P, typename CONTAINER>
+requires std::regular_invocable<P, CONTAINER, std::size_t> bool
+is_contiguous_on(P&& projection, const CONTAINER& in) {
+
+    // Grab the number of elements in our container.
+    std::size_t n = in.size();
```

Comment:
No, I don't like this. This could stay the same with a proxy type; let's not go around making code less functional and harder to read to facilitate this change if it can be avoided.